### PR TITLE
[WIP] integrate export-to-csv directive into ng-admin

### PR DESCRIPTION
**Pull request by ThieryMichel**
feature: #230 
- [x] integrate export-to-csv into ng-admin @fzaninotto @manuquentin what do you think?
- [x] allow to configure the view used for the fields
- [x] use current list order for the export
- [ ] add test

### CODE REVIEW

**Comment by endel** [comment]
Hi @ThieryMichel, I'm trying to test your branch, and the following error is occurring:

```
Unknown provider: clEntryFormaterProvider <- clEntryFormater <- maExportToCsvButtonDirective
```

Am I missing something?

**Comment by ThieryMichel** [comment]
No it was a mistake on my part. Corrected it.

File: src/javascripts/ng-admin/Crud/misc/EntryFormater.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                case 'email':
+                case 'json':
+                case 'date':
+                case 'choice':
+                case 'choices':
+                case 'file':
+                case 'template':
+                    map[order] = {name: attr, type: 'Field'};
+                    break;
+                case 'Reference':
```
Comment (fzaninotto) [inline src/javascripts/ng-admin/Crud/misc/EntryFormater.js]: must be `reference` with a lowercase `c`. Same for `referenced_list` (refs #307).

**Comment by fzaninotto** [comment]
Very nice. For the configurability, I suggest that you use a smart setter comparable to `fields()`:

```js
// by default, the csv export should use the field list from the list view
myEntity.listView()
  .exportFields(myEntity.listView.fields())
// developers can customize it with a custom field list
myEntity.listView()
  .exportFields([
    nga.field('foo'),
    nga.field('bar')
  ])
```

**Comment by fzaninotto** [comment]
Bump

**Comment by lucval** [comment]
Hi @ThieryMichel, is there any commit to solve the problem pointed out by @endel on February 19?
I am still getting this:
"Unknown provider: clEntryFormaterProvider <- clEntryFormater <- maExportToCsvButtonDirective"

Thanks in advance

**Comment by ThieryMichel** [comment]
Just fixed it.

**Comment by fzaninotto** [comment]
Thanks for this great feature!

**Pull request merged**